src/controllers/DataSetAdmin.js

```diff
@@ -0,0 +1,250 @@
+        val = null
+      }
+      row[key] = val
+    })
+    row.dataset = dataSetId
+    //nb: must return changed row rather than new object
+    return row
+  }
+
+
+   /**
+    * override EditController save for datasets-dataset:
+   * handle POST response for 
+   * editing an existing 
+   * or creating a new dataset
+   */
+  save(req, res) {
+    let dataSetId = req.params.id
+    this.log.debug("save DataSet id: ", dataSetId)
+    return Promise.resolve().then( () => {
+      if (!dataSetId) {
+        if (!req.body.name) {
+          throw new Error('DataSet name is required.')
+        } else {
+          return this.model.create({name: req.body.name, source: req.body.source, description: req.body.description})
+          .then( (newRecord) => {
+          this.log.debug("dataset created ", newRecord.id)
+          return res.redirect(this.routePrefix)
+          })
+        }
+      } else {
+        this.model.findOne(dataSetId)
+        .then((dataset) => {
+          dataset.name = req.body.name
+          dataset.source = req.body.source
+          dataset.description = req.body.description
+          _.each( dataset.fields, (dsField) => {
+            dsField.isVisible = false;
+            dsField.isPrimaryKey = false;
+            _.each(dsField, (value,key) => {
+              let formValue = req.body['field.' + key + '.' + dsField.id]
+              if (formValue) {
+                if (key.startsWith("is")) { //boolean
+                  dsField[key] = ("true"==formValue)
+                } else {
+                  dsField[key] = formValue
+                } 
+              }
+              //field.isPrimaryKey in body is set to field.id
+              if (dsField.id == req.body['field.isPrimaryKey']) {
+                dsField.isPrimaryKey = true
+              }
+            })
+          })
+          return dataset
+        }).then( (dataset) => {
+          return dataset.save()
+        }).then( () => {
+          return res.redirect(this.routePrefix)
+        })
+      }
+    }).catch( (err) => {
+      req.flash('error', 'Unable to create or edit: ' + err.message)
+      return res.redirect(this.routePrefix)
+    })
+  }
+
+  /**
+   * Handler for removing dataset.
+   * Deleting a dataset must also delete child datarows
+   */
+  remove(req, res) {
+    let dataSetId = req.param('id');
+    return this.models['datasets-datarow'].destroy({dataset: dataSetId}) 
+    .then( (rowsRemoved) => {
+      this.log.debug("remove datasets-dataset deleted ", rowsRemoved.length, " data rows from dataset ", dataSetId); 
+      req.flash('info', "Successfully removed "+rowsRemoved.length+" data rows")
+      return this.model.destroy({id: dataSetId}).then( (dsRemoved) => {
+        this.log.debug("removed DataSet ", dataSetId); 
+        res.redirect(this.routePrefix)
+      })
+    }).catch((e) => {
+      this.log.debug( "nxus-dataset error removing data rows from dataset " + dataSetId, e);

```

Comment:
Probably want this to use `log.error` so it'll show up on a production environment.